qemu/tests/hotplug_mem_repeat.py

```diff
@@ -7,64 +6,68 @@
 
 class MemoryHotplugRepeat(MemoryHotplugTest):
 
+    def repeat_hotplug(self, vm, target_mems):
+        """
+        Hotplug memory in target_mems
+        :param vm: vm object in this test
+        :param target_mems: target memory to be hotplugged
+        """
+        for target_mem in target_mems.split():
+            self.hotplug_memory(vm, target_mem)
+        self.check_memory(vm)
+
+    def repeat_unplug(self, vm, target_mems):
+        """
+        Unplug memory in target_mems
+        :param vm: vm object in this test
+        :param target_mems: target memory to be unplugged
+        """
+        for target_mem in target_mems.split():
+            self.unplug_memory(vm, target_mem)
+        self.check_memory(vm)
+
     def start_test(self):
         """
-        Prepare reqired test params, then start memory
-        hotplug/unplug tests in turn. And this is test entry.
+        Prepare required test params, then for scalability test, hotplug
+        memory 256 times, then unplug 256 times. Otherwise, repeat hotplug
+        and unplug in turn for 256 times. This is test entry.
         """
-        target_mem = self.params["target_mem"]
+        times = int(self.params["repeat_times"])
+        target_mems = ""
+        for i in range(times):
+            target_mems += "mem%s " % i
         vm = self.env.get_vm(self.params["main_vm"])
-        max_slots = int(self.params.get("slots_mem", 4))
-        scalability_test = self.params.get("scalability_test") == "yes"
-        repeats = scalability_test and max_slots or self.params["repeats"]
         original_mem = self.get_guest_total_mem(vm)
-        for repeat in range(int(repeats)):
-            extra_params = (scalability_test and
-                            [{'slot_dimm': repeat}] or [None])[0]
-            error_context.context("Hotplug/unplug loop '%d'" % repeat,
-                                  logging.info)
-            self.turn(vm, target_mem, extra_params)
+        if self.params["test_type"] == "scalability_test":
+            error_context.context("Repeat hotplug memory for %s times"
+                                  % times, logging.info)
+            self.repeat_hotplug(vm, target_mems)
+            if self.params.get('os_type') == 'linux':
+                error_context.context("Repeat unplug memory for %s times"
+                                      % times, logging.info)
+                self.repeat_unplug(vm, target_mems)
+        else:
+            for target_mem in target_mems.split():
```

Comment:
We can convert `target_mems` to a list after it is generated, then no need to convert it for again in `repeat_hotplug`, `repeat_unplug` .

Reply:
Good suggestion! Have updated to initialize it as a list.